src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Definition/RazorDefinitionEndpoint.cs

```diff
@@ -128,12 +131,81 @@ public DefinitionRegistrationOptions GetRegistrationOptions(DefinitionCapability
                 new LocationOrLocationLink(new Location
                 {
                     Uri = originComponentUri,
-                    Range = new Range(new Position(0, 0), new Position(0, 0)),
+                    Range = range,
                 }),
             });
         }
 
-        internal static async Task<TagHelperBinding?> GetOriginTagHelperBindingAsync(
+        private async Task<Range> GetNavigateRangeAsync(DocumentSnapshot documentSnapshot, BoundAttributeDescriptor? attributeDescriptor, CancellationToken cancellationToken)
+        {
+            if (attributeDescriptor is not null)
+            {
+                var originCodeDocument = await documentSnapshot.GetGeneratedOutputAsync().ConfigureAwait(false);
+                var range = await TryGetPropertyRangeAsync(originCodeDocument, attributeDescriptor.GetPropertyName(), _documentMappingService, cancellationToken).ConfigureAwait(false);
+
+                if (range is not null)
+                {
+                    return range;
+                }
+            }
+
+            return new Range(new Position(0, 0), new Position(0, 0));
+        }
+
+        internal static async Task<Range?> TryGetPropertyRangeAsync(RazorCodeDocument codeDocument, string propertyName, RazorDocumentMappingService documentMappingService, CancellationToken cancellationToken)
+        {
+            // Parse the C# file and find the property that matches the name, and that has a [Parameter] attribute
+
+            var csharpText = codeDocument.GetCSharpSourceText();
+            var syntaxTree = CSharpSyntaxTree.ParseText(csharpText, cancellationToken: cancellationToken);
+            var root = await syntaxTree.GetRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var property = root.DescendantNodes(n => n is BaseNamespaceDeclarationSyntax or CompilationUnitSyntax or ClassDeclarationSyntax)
```

Comment:
OH so interesting! Is this what we do on the Roslyn side too or are these types of inspections abstracted behind helper services?

Reply:
The Roslyn equivalent here uses symbols, so kinda. Symbols are a helper in terms of finding attributes because the compiler nicely flattens all lists, and resolves types and aliases etc.

Reply:
I c i c. I wonder what something like that would look like on our side. I guess I wish the Razor compiler output C# syntax trees as their C# generated document instead of big ass strings 😆 (/cc @javiercn)